### PR TITLE
storage: Remove MVCC value allocation on Put

### PR DESCRIPTION
**Pull request by nvb**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5553)

### CODE REVIEW

File: storage/engine/mvcc.go [hunk 1]
```diff
@@ -1047,7 +1045,7 @@ func MVCCIncrement(engine Engine, ms *MVCCStats, key roachpb.Key, timestamp roac
 	defer iter.Close()
 
 	var int64Val int64
-	err := mvccPutUsingIter(engine, iter, ms, key, timestamp, nil, txn, func(value *roachpb.Value) ([]byte, error) {
+	err := mvccPutUsingIter(engine, iter, ms, key, timestamp, noValue, txn, func(value *roachpb.Value) ([]byte, error) {
```
Comment (tschottdorf) [inline storage/engine/mvcc.go]: I've been thinking about this pattern as well. Could save all of these random inline comments - `noTxn`, ....

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5553#-:-KDeMFvroEXTpsq_Pzxd:1417088721)*

**Comment by RaduBerinde** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5553#-:-KDeXbRKgdtaSikofTnV:1417088721)*

File: storage/engine/mvcc.go (hunk 1 above)
Comment (spencerkimball) [inline storage/engine/mvcc.go]: Yeah, it's a great pattern.

**Pull request merged**